System.Compiler/Writer.cs

```diff
@@ -2780,6 +2780,9 @@ class MethodSemanticsRowComparer : IComparer{
           }
           this.IncrementStackHeight();
           return;
+        case NodeType.This:
+          this.methodBodyHeap.Write((byte)0x02);
```

Comment:
This fix is so small! But it affects behavior in such a drastic way:)

Looks good to me! Great job!